packages/esm-form-render-app/src/render/forms-render-test.component.tsx

```diff
@@ -74,6 +75,65 @@ function FormRenderTest() {
     setIsSchemaLoaded(false);
   };
 
+  const formValidator = () => {
+    if (defaultJson) {
+      const parsedForm = typeof defaultJson == 'string' ? JSON.parse(defaultJson) : defaultJson;
+
+      for (let i = 0; i < parsedForm.pages.length; i++) {
+        for (let j = 0; j < parsedForm.pages[i].sections.length; j++) {
+          for (let k = 0; k < parsedForm.pages[i].sections[j].questions.length; k++) {
+            
+            const questionObject = parsedForm.pages[i].sections[j].questions[k];
+            handleFormValidation(questionObject);
+          }
+        }
+      }
+    } else {
+      console.log('Empty form!');
+    }
+  };
+
+  const handleFormValidation = (conceptObject) => {
+    conceptObject.questionOptions.concept
+      ? openmrsFetch(`/ws/rest/v1/concept/${conceptObject.questionOptions.concept}`)
+          .then((response) => {
+            console.log(response.data);
+            dataTypeChecker(conceptObject, response);
+          })
+          .catch((error) => {
+            error.message.includes('404') &&
+              console.log(`❌ Concept UUID ${conceptObject.questionOptions.concept} not found`);
+          })
+      : console.log('❌ Question with no concept UUID: ', conceptObject.id);
+  };
+
+  const dataTypeChecker = (conceptObject, responseObject) => {
+    const dataTypes = ['Numeric', 'Coded', 'Text', 'Date', 'Datetime', 'Boolean', 'Rule'];
+
+    const renderTypes = [
+      ['number'],
+      ['select', 'checkbox', 'radio', 'toggle', 'content-switcher'],
+      ['text', 'textarea'],
+      ['date'],
+      ['datetime'],
+      ['toggle', 'select', 'radio', 'content-switcher'],
+      ['repeating', 'group']
+    ];
+
+    if (conceptObject.questionOptions.concept === responseObject.data.uuid) {
```

Comment:
Do you expect to have a response that is different than the request `conceptUuid` requested? I think yes and you already taking care of that on your `catch`, therefore I don't think this if is necessary. And if this ever happen it would mean a fundamental bug on the backend, you can still check (defensive code) but not on this function for the sake of separation of concerns.  But feel free to comment.

Reply:
The if actually isn't that necessary, I just added it for precaution, will resolve.

Reply:
I think you can still do it but outside of this function, do right after getting the response and only invoke this method if it is a pure match otherwise the backend is broken.

Reply:
This is done.